packages/core/src/animation/Animator.ts

```diff
@@ -618,4 +650,21 @@ export class Animator extends Component {
       }
     }
   }
+
+  private _fireAnimationEvents(eventHandlers: AnimationEventHandler[], lastClipTime: number, clipTime: number) {
+    for (let i = eventHandlers.length - 1; i >= 0; i--) {
+      const eventHandler = eventHandlers[i];
+      const { time, parameter } = eventHandler.event;
+
+      if (time > clipTime) break;
+      if (time < lastClipTime) continue;
+
+      const { handlers } = eventHandler;
+      if (time === MathUtil.clamp(time, lastClipTime, clipTime)) {
+        for (let j = handlers.length - 1; j >= 0; j--) {
+          handlers[j](parameter);
+        }
+      }
+    }
+  }
```

Comment:
You can do some simple optimizations, for example store current event index to reduce loop count.

Reply:
done